### PR TITLE
TrinoHook add authentication via JWT token and Impersonation

### LINKED ISSUE #18969: Trino JWT Authentication Support
### Description

Would be great to support JWT Authentication in Trino Hook (which also can be used for presto hook).

For example like this

```
        elif extra.get('auth') == 'jwt':
            auth = trino.auth.JWTAuthentication(
                token=extra.get('jwt__token')
            )
```

### Use case/motivation

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by pragyak412**
closes: https://github.com/apache/airflow/issues/18969
This change introduces the authentication method via JWT and also the possibility to impersonate as DAG owner.
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
You need to fix static checks (And rebase as some unrelated errors have been fixed).

**Comment by pragyak412** [comment]
@potiuk Can you suggest anything that has to be changed now?

File: tests/providers/trino/hooks/test_trino.py [hunk 1]
```diff
@@ -82,6 +83,21 @@ def test_get_conn_kerberos_auth(self, mock_get_connection, mock_connect, mock_au
         )
         TrinoHook().get_conn()
         self.assert_connection_called_with(mock_connect, auth=mock_auth)
+
```
Comment (eladkal) [inline tests/providers/trino/hooks/test_trino.py]:
```suggestion

```

File: airflow/providers/trino/hooks/trino.py [hunk 2]
```diff
@@ -81,10 +84,14 @@ def get_conn(self) -> Connection:
                 delegate=_boolify(extra.get('kerberos__delegate', False)),
                 ca_bundle=extra.get('kerberos__ca_bundle'),
             )
+        
+        if _boolify(extra.get('impersonate_as_owner', False)):
+            user = os.getenv('AIRFLOW_CTX_DAG_OWNER', None)
+
```
Comment (eladkal) [inline airflow/providers/trino/hooks/trino.py]:
```suggestion

```

File: airflow/providers/trino/hooks/trino.py [hunk 3]
```diff
@@ -81,10 +84,14 @@ def get_conn(self) -> Connection:
                 delegate=_boolify(extra.get('kerberos__delegate', False)),
                 ca_bundle=extra.get('kerberos__ca_bundle'),
             )
+
```
Comment (eladkal) [inline airflow/providers/trino/hooks/trino.py]:
```suggestion

```

**Review by eladkal — CHANGES_REQUESTED** [review]
Need to remove white spaces to fix static checks

File: airflow/providers/trino/hooks/trino.py [hunk 4]
```diff
@@ -81,10 +84,12 @@ def get_conn(self) -> Connection:
                 delegate=_boolify(extra.get('kerberos__delegate', False)),
                 ca_bundle=extra.get('kerberos__ca_bundle'),
             )
+        if _boolify(extra.get('impersonate_as_owner', False)):
+            user = os.getenv('AIRFLOW_CTX_DAG_OWNER', None)
```
Comment (uranusjr) [inline airflow/providers/trino/hooks/trino.py:122]: Should this fall back to `db.login` if `AIRFLOW_CTX_DAG_OWNER` is not set?

**Review by potiuk — CHANGES_REQUESTED** [review]
Needs documentation update.

**Comment by eladkal** [comment]
Can you please fix static checks?
https://github.com/apache/airflow/blob/main/BREEZE.rst#running-static-checks

**Comment by eladkal** [comment]
@pragyak412 can you fix the test and resolve conflicts?

**Comment by pragyak412** [comment]
> @pragyak412 can you fix the test and resolve conflicts?

>> SURR!

**Comment by pragyak412** [comment]
> @pragyak412 can you fix the test and resolve conflicts?

Sure!

**Comment by eladkal** [comment]
static tests are failing. Can you please fix it?

**Comment by pragyak412** [comment]
> static tests are failing. Can you please fix it?

I ran the tests locally. I have fixed the documentation.  
Screenshot 2022-04-20 at 1 34 27 PM

But the static test is getting passed locally for files.
Screenshot 2022-04-20 at 1 00 23 PM

**Pull request closed without merging**